compiler/test/dotty/tools/repl/ReplTest.scala

```diff
@@ -56,26 +57,22 @@ class ReplTest(withStaging: Boolean = false, out: ByteArrayOutputStream = new By
 
   def testFile(f: JFile): Unit = {
     val prompt = "scala>"
-    val lines = Source.fromFile(f, "UTF-8").getLines().buffered
 
-    assert(lines.head.startsWith(prompt),
-           s"""Each file has to start with the prompt: "$prompt"""")
-
-    def extractInputs(prompt: String): List[String] = {
+    def extractInputs(lines: BufferedIterator[String]): List[String] = {
```

Comment:
Could this use a plain collection like a Seq instead of an Iterator? I find iterators way too easy to misuse (e.g., if I try to print their content, then I break any code afterward), and this code isn't performance critical.

Reply:
Sure, I'll improve this :+1: